cmd/controller-manager/app/options/options.go

```diff
@@ -38,16 +38,16 @@ const (
 
 // ServerOption is the main context object for the controllers.
 type ServerOption struct {
-	KubeClientOptions    kube.ClientOptions
-	CertFile             string
-	KeyFile              string
-	CaCertFile           string
-	CertData             []byte
-	KeyData              []byte
-	CaCertData           []byte
-	EnableLeaderElection bool
```

Comment:
How about keeping the old one for better compatibility?

Reply:
`leader-elect` flag is kept when using BindLeaderElectionFlags.

Reply:
> How about keeping the old one for better compatibility?
Use leaderElection flagSet instead, it's a leaderElectionOption field.

Reply:
But `lock-object-namespace` flag seems has been removed.

Reply:
> But lock-object-namespace flag seems has been removed.

use the `leader-elect-resource-namespace`  flag instead

Reply:
So plz also update params in `helm` : )

Reply:
I think we should better keep the old flag and mark it deprecated first.

Reply:
> So plz also update params in helm  I think we should better keep the old flag and mark it deprecated first.

sure!

Reply:
> I think we should better keep the old flag and mark it deprecated first.

I also means keeping the old one and marking it is deprecated in recent release.